agent/structs/config_entry_routes.go

```diff
@@ -432,6 +435,18 @@ type URLRewrite struct {
 	Path string
 }
 
+type RetryFilter struct {
+	NumRetries            *uint32
+	RetryOn               []string
+	RetryOnStatusCodes    []uint32
+	RetryOnConnectFailure *bool
+}
+
+type TimeoutFilter struct {
+	RequestTimeout time.Duration
+	IdleTimeout    time.Duration
```

Comment:
Since we use `time.Duration` instead of `*time.Duration`, what happens if we set one but not the other? I suspect the pre-existing code handles this but wanted to be sure.

Reply:
Setting *time.Duration caused several failures in the autogenerated functions. I'll add a test to cover that case so we know what happens.

Reply:
An empty time.Duration is not set in the XDS config. 👍